### ISSUE TITLE
Fine-tuned AD checkpoints do not replay training-time preprocessing

### ISSUE THREAD

**Issue by PansulBhatt** [NVIDIA/NV-Tesseract#31]
## Summary

Fine-tuned AD checkpoints save training-time preprocessing metadata, but SDK inference does not consume it. Instead, inference fits a new dimensionality transform and min-max scaler on each inference batch.

That gives the trained model a different feature representation from the one it saw during fine-tuning.

## Why this matters

Fine-tuning fits `FeatureAdapter` on the training split and uses those statistics for both training and validation. The checkpoint currently stores `preprocessing`, but `inference_ad_tesseract2()` only reads `config` and `model`; it then calls the generic dataloader, which re-runs `match_target_dim()` and `min_max_normalize()` on the inference data.

This has several effects:

- The same observation can receive different model inputs depending on the other rows in its inference batch.
- A distribution shift can be normalized back into the apparent training range, weakening anomaly evidence.
- PCA is re-fit on inference data instead of using the training projection.
- Existing checkpoint metadata records whether PCA was used, but not the fitted PCA components or mean, so PCA-based fine-tuning cannot be replayed.
- Numeric metadata or label columns can alter feature order or dimensionality unless callers manually remove them.

## Minimal reproduction

With a scale factor of 20, fit preprocessing on training values in the range `0..10`, then transform serving values in the range `100..110`:

```text
training-time adapter output:
[[200. 200.]
 [210. 210.]
 [220. 220.]]

current SDK batch-fitted output:
[[ 0.  0.]
 [10. 10.]
 [20. 20.]]
```

The difference is not cosmetic: the first tensor preserves the serving data's distance from the training distribution, while the second re-scales that shifted batch to look in-range.

## Proposed solution

1. Move feature adaptation into a shared SDK module used by both fine-tuning and inference.
2. Serialize the complete fitted transform in fine-tuned checkpoints:
   - input and target dimensions
   - fitted dtype
   - scale factor
   - min/max statistics
   - PCA components and PCA mean when PCA is used
   - feature column names and order
   - window length and mask split
3. When no explicit external `preprocess_model_dir` is supplied, automatically replay checkpoint preprocessing in both single- and multi-GPU inference paths.
4. Keep the existing generic preprocessing path unchanged for pretrained checkpoints without embedded metadata.
5. Fail clearly for malformed metadata or legacy PCA checkpoints whose fitted PCA state was never stored; silently re-fitting PCA would produce incorrect inputs.

## Suggested regression coverage

- Compare replayed checkpoint preprocessing to the original fitted adapter on shifted serving data.
- Verify PCA replay numerically matches the original fitted transform.
- Save and reload a real fine-tune checkpoint, then confirm the transform survives serialization.
- Verify reordered columns and extra numeric labels do not change model inputs.
- Verify missing training columns fail clearly.
- Verify pretrained checkpoints without metadata and explicit external preprocessors keep their existing behavior.

This is a preprocessing correctness issue. Downstream ROC-AUC/F1 impact should be measured separately with a controlled old-vs-new inference A/B using the same fine-tuned checkpoint, labeled holdout data, seeds, diffusion settings, and threshold calibration.

**Comment by PansulBhatt** [comment]
Created this for reference on thoughts around how to resolve - https://github.com/NVIDIA/NV-Tesseract/pull/32 for ref
